src/main/java/com/jhipster/compositekey/service/dto/BusinessBasicIndexDTO.java

```diff
@@ -95,30 +97,37 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) {
             return false;
         }
-
         BusinessBasicIndexDTO businessBasicIndexDTO = (BusinessBasicIndexDTO) o;
-        return Objects.equals(businessId, businessBasicIndexDTO.businessId)
-            && Objects.equals(basicIndexId, businessBasicIndexDTO.basicIndexId)
-            && Objects.equals(year, businessBasicIndexDTO.year)
-            ;
-
+        if (businessBasicIndexDTO.getBusinessId() == null && businessBasicIndexDTO.getBasicIndexId() == null && businessBasicIndexDTO.getYear() == null) {
+            return false;
+        }
+        if (getBusinessId() == null && getBasicIndexId() == null && getYear() == null) {
+            return false;
+        }
+        return Objects.equals(getBusinessId(), businessBasicIndexDTO.getBusinessId())
+                && Objects.equals(getBasicIndexId(), businessBasicIndexDTO.getBasicIndexId())
+                && Objects.equals(getYear(), businessBasicIndexDTO.getYear());
     }
 
     @Override
     public int hashCode() {
         int result = 17;
-        result = 31 * result + businessId.hashCode();
-        result = 31 * result + basicIndexId.hashCode();
-        result = 31 * result + year.hashCode();
+        result = 31 * result + Objects.hashCode(this.businessId);
+        result = 31 * result + Objects.hashCode(this.basicIndexId);
+        result = 31 * result + Objects.hashCode(this.year);
         return result;
     }
 
     @Override
     public String toString() {
         return "BusinessBasicIndexDTO{" +
+            "businessId=" + getBusinessId() +
```

Comment:
```
    @Override
    public String toString() {
        return "BusinessBasicIndexDTO{" +
            ", businessId=" + businessId +
            ", basicIndexId=" + basicIndexId +
            ", year=" + year +
            ", month=" + month +
            ", value=" + value +
            "}";
    }
```
I don't now if we should include the names, to discuss...

Reply:
I add same question when impleting those changes. I think it is already a bug for dto since it is added fields name from  foreign object but not included in the toString.